code/atom.dm

```diff
@@ -690,11 +690,13 @@
 
 //mbc : sorry, i added a 'is_special' arg to this proc to avoid race conditions.
 /atom/proc/attackby(obj/item/W as obj, mob/user as mob, params, is_special = 0)
+	if (SEND_SIGNAL(src, COMSIG_ATOM_ATTACK_BY, W, user, params, is_special) & COMSIG_RETURN_EARLY)
+		return
```

Comment:
Similar concerns to attack_hand parent-calling apply here. It's probably currently okayish as-is but it might be a good idea to use some sort of COMSIG_ATOM_PRE_ATTACK_BY signal that gets called *before* attackby itself and can cancel attackby. There don't seem to be that many places that call attackby so it's probably feasible.

Reply:
I'll take okayish for now!